pkg/providers/snow/defaults.go

```diff
@@ -70,6 +84,21 @@ func (md *MachineConfigDefaulters) SetupDefaultSshKey(ctx context.Context, m *v1
 		return nil
 	}
 
+	keyCount, err := md.defaultKeyCount(ctx)
+	if err != nil {
+		return err
+	}
+
+	if keyCount > 0 && keyCount < len(md.defaulters.awsClientMap) {
+		return fmt.Errorf("default key [keyName=%s] only exists on some of the devices. Use 'aws ec2 import-key-pair' to import this key to all the devices", defaultAwsSshKeyName)
```

Comment:
nit: shouldn't we ideally just import it wherever is needed? I'm ok if you want that as an improvement later

Reply:
we don't have access to the public key tho. Thus we can't import for the user

Reply:
will follow up on this